src/Illuminate/Mail/MailManager.php

```diff
@@ -120,8 +120,28 @@ protected function resolve($name)
         // Once we have created the mailer instance we will set a container instance
         // on the mailer. This allows us to resolve mailer classes via containers
         // for maximum testability on said classes instead of passing Closures.
+        $mailer = $this->build(['name' => $name, ...$config]);
+
+        // Next we will set all of the global addresses on this mailer, which allows
+        // for easy unification of all "from" addresses as well as easy debugging
+        // of sent messages since these will be sent to a single email address.
+        foreach (['from', 'reply_to', 'to', 'return_path'] as $type) {
+            $this->setGlobalAddress($mailer, $config, $type);
+        }
+
+        return $mailer;
+    }
+
+    /**
+     * Build a new mailer instance.
+     *
+     * @param  array  $config
+     * @return \Illuminate\Mail\Mailer
+     */
+    public function build($config)
+    {
         $mailer = new Mailer(
-            $name,
+            $config['name'] ?? 'ondemand',
```

Comment:
We use `ondemand` as the driver name in the `FilesystemManager`:

https://github.com/laravel/framework/blob/0be1995727a7c9341d13a08253a12140b8fc84fd/src/Illuminate/Filesystem/FilesystemManager.php#L106